packages/core/src/service/Project.ts

```diff
@@ -395,16 +398,30 @@ export class Project implements ExternalEventEmitter {
 		const loadConfig = async () => {
 			this.config = await this.#configService.load()
 			this.ignore = ignore()
+			await parseExcludePaths()
+			for (const pattern of this.#excludePaths) {
+				this.ignore.add(pattern)
+			}
+		}
+
+		const parseExcludePaths = async () => {
+			const paths = []
 			for (const pattern of this.config.env.exclude) {
 				if (pattern === '@gitignore') {
 					const gitignore = await this.readGitignore()
 					if (gitignore) {
-						this.ignore.add(gitignore)
+						const gitignoreLines = gitignore.split(/\r?\n/)
+						const gitignorePaths = gitignoreLines
+							.filter((line) => line !== '' && !line.startsWith('#'))
+						paths.push(...gitignorePaths)
 					}
 				} else {
-					this.ignore.add(pattern)
+					paths.push(pattern)
 				}
 			}
+			// Chokidar never matches paths with a trailing (back)slash, so fix any paths
+			// that may be specified as such
+			this.#excludePaths.push(...paths.map((path) => path.replace(/(\\|\/)$/, '')))
```

Comment:
Since this is a chokidar specific requirement, this should be moved to the `NodeJsExternals` `watch` handler.